ntoskrnl/io/iomgr/deviface.c

```diff
@@ -28,6 +29,257 @@ IopGetDeviceObjectFromDeviceInstance(PUNICODE_STRING DeviceInstance);
 
 static PWCHAR BaseKeyString = L"\\Registry\\Machine\\System\\CurrentControlSet\\Control\\DeviceClasses\\";
 
+/**
+ * @brief IopBuildSymbolicLink
+ * 
+ * Creates a new symbolic link from the specified prefix, device string,
+ * class GUID and reference string (if any).
+ * 
+ * @param[in] PrefixString
+ * Prefix of symbolic link. '\??\' for Kernel mode or '\\?\' for User mode.
+ * 
+ * @param[in] DeviceString
+ * Device string, placed after prefix and before GUID, looks like ACPI#PNP0501#1#.
+ * 
+ * @param[in] GuidString
+ * Device interface class GUID represented by a string. Placed in curly brackets {},
+ * after device string, should always be 38 characters long. Looks like
+ * {01234567-89ab-cdef-0123-456789abcdef}.
+ * 
+ * @param[in,opt] ReferenceString
+ * Optional reference string, if any. Placed after GUID, at the end of symbolic link.
+ * Usually contains human-readable subdevice name or class GUID.
+ * 
+ * @param[out] SymbolicLinkName
+ * Pointer to unicode string which receives created symbolic link.
+ * 
+ * @return
+ * STATUS_SUCCESS in case of success, error NTSTATUS code otherwise.
+ *
+ */
+static
+NTSTATUS
+IopBuildSymbolicLink(
+    _In_ PCUNICODE_STRING PrefixString,
+    _In_ PCUNICODE_STRING DeviceString,
+    _In_ PCUNICODE_STRING GuidString,
+    _In_opt_ PCUNICODE_STRING ReferenceString,
+    _Out_ PUNICODE_STRING SymbolicLinkName)
+{
+    UNICODE_STRING PathSep = RTL_CONSTANT_STRING(L"\\");
```

Comment:
```suggestion
    const UNICODE_STRING PathSep = RTL_CONSTANT_STRING(L"\\");
```

Reply:
Thanks. Added a constant identifier, same as for other constant strings.